lib/grape/endpoint.rb

```diff
@@ -231,7 +231,7 @@ def run
 
         run_filters before_validations, :before_validation
 
-        run_validators validations, request unless @method_not_allowed
+        run_validators validations, request unless env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]
```

Comment:
You might want to audit the use if `@variable` elsewhere, we have at least an `@app_response` in a formatter middleware, `@named_params` in a DSL helper and `@versions` in routing.